src/app/components/SearchForm/SearchForm.jsx

```diff
@@ -34,9 +34,17 @@ class SearchForm extends React.Component {
     if (!this.state.searchField) {
       this.setState({ searchField: 'q' });
     }
+    if (!this.state.searchQuery) {
+      throw new Error('Please enter a term or terms to search');
+    }
 
     const terms = this.state.searchQuery.trim().replace(/\s+/g, ' ');
-    this.props.search(terms, this.state.searchField);
+
+    if (!this.state.searchField || this.state.searchField === 'q') {
+      this.props.searchGet(terms);
+    } else {
+      this.props.searchPost(terms, this.state.searchField);
```

Comment:
Why the two different actions?

Reply:
At present, we're using searchGet method for general keyword searching across available fields and the searchPost method to handle field specific and advanced search queries which specify one or more fields with filtering and aggregations. The search API is fairly simple for now and doesn't do more sophisticated parsing like the Discovery API might be doing with its more complicate GET requests.